assets/src/edit-story/components/helpCenter/useHelpCenter.js

```diff
@@ -102,6 +135,52 @@ export function useHelpCenter() {
     [store.actions]
   );
 
+  const getReadTips = useCallback(async () => {
+    try {
+      actions.fetchReadStart();
+      const { meta } = await getCurrentUser();
+
+      actions.fetchReadSuccess({
+        read: meta.web_stories_onboarding ?? [],
+      });
+    } catch {
+      actions.fetchReadError();
+    }
+  }, [actions, getCurrentUser]);
+
+  useEffect(() => {
+    getReadTips();
+  }, [getReadTips]);
+
+  const updateReadStatus = useCallback(async () => {
+    const currentKey = store.state.navigationFlow[store.state.navigationIndex];
+    const [doneKey] = DONE_TIP_ENTRY;
+    // if the current status is not 'read'
+    if (
+      currentKey &&
+      store.state.read.indexOf(currentKey) === -1 &&
+      currentKey !== doneKey
+    ) {
+      try {
+        // optimistically mark it read
+        // whether the update call succeeds or not
+        actions.read(currentKey);
+        await updateCurrentUser({
+          meta: {
+            web_stories_onboarding: [...store.state.read, currentKey],
+          },
+        });
+        actions.readSuccess();
+      } catch (e) {
+        actions.readError(currentKey);
+      }
+    }
+  }, [store, actions, updateCurrentUser]);
+
+  useEffect(() => {
+    updateReadStatus();
+  }, [updateReadStatus, store.state.navigationIndex]);
```

Comment:
These feel like two separate things kinds confounded into 1.

I think having to add `store.state.navigationIndex` is kind of an indication of this.

I think the two separate actions should be to:
1) update the state to read the current key
2) persist all the read objects

Item one should just be handled by the store action, no need for a Callback in here.

Item 2 should be an effect that runs everytime the read tips in the store state update.
```
useEffect(() => {
  const asyncActionToPersistState = async () => {...};
  await asyncActionToPersistState();
}, [state.store.readTips])
```